core/src/components/pagination/pagination.ts

```diff
@@ -448,6 +448,15 @@ export function createPagination(config?: PropsConfig<PaginationProps>): Paginat
 		};
 	});
 
+	function handleNavigation(pageNumber: number, event?: Event & {ctrlKey: boolean; metaKey: boolean}) {
+		if (!event || (!event.ctrlKey && !event.metaKey)) {
+			setTimeout(() => patch({page: pageNumber}));
```

Comment:
do we need the setTimeout here ?

Reply:
it is needed only for 'next' and 'previous' just because the patch it is changing the urls, before the browser is changing the url when pressing the `a`.
as example
initial state: `currentPage: 4`, `nextHref: #{page:5}`, browser url: url#{page:4}
click on 'next' -> `currentPage: 5`, `nextHref: #{page:6}`, browser url: url#{page:6} -instead of 5
skipping one tick it's updating the browser href, before the current page which updates the next

Reply:
Good catch for this issue ! 

I don't know if it's a good solution or not, and what kind of checks that need to be done, but the only way I found to keep it synchronous is finally to manage the href internally, like this:

```typescript
		if (!event || (!event.ctrlKey && !event.metaKey)) {
			patch({page: pageNumber});
			const href = (event?.target as HTMLAnchorElement)?.href;
			if (href) {
				history.pushState(null, '', href);
				event?.preventDefault();
			}
		}

```

Reply:
Thanks :) I'll give it a try .